cmd/lncli/cmd_invoice.go

```diff
@@ -80,10 +85,9 @@ func addInvoice(ctx *cli.Context) error {
 
 	args := ctx.Args()
 
-	switch {
-	case ctx.IsSet("amt"):
-		amt = ctx.Int64("amt")
-	case args.Present():
+	amt = ctx.Int64("amt")
+	amtMsat = ctx.Int64("amt_msat")
+	if !ctx.IsSet("amt") && !ctx.IsSet("amt_msat") && args.Present() {
```

Comment:
Since `lnrpc.Invoice.Value` and `lnrpc.Invoice.ValueMsat` are mutually exclusive, I'd check here that only one of them is set and print errors accordingly.

Reply:
@bhandras IIRC the issue was discussed related to my previous addhodlinvoice patch, and deferring error detection to lnd-side lnrpc.UnmarshallAmt was deemed to be okay. Are you sure we want an inconsistency here? Maybe the addholdinvoice patch should also be revised along the same line?

Reply:
Indeed `lnrpc.UnmarshallAmt` handles error detection. I agree it's fine handling the error there.